CMakeLists.txt

```diff
@@ -442,6 +443,26 @@ set (UPM_COMMON_HEADER_DIRS ${CMAKE_HOME_DIRECTORY}/include)
 # Generate a build-only C++ header to add functionality to SWIG'ed modules
 configure_file (${PROJECT_SOURCE_DIR}/cmake/modules/version.hpp.in ${PROJECT_BINARY_DIR}/src/version.hpp @ONLY)
 
+# Cordova binding
+if (BUILDCORDOVABINDING)
+  if (BUILDSWIGJAVA)
+  else()
+    message(FATAL_ERROR "Must enable java binding first")
```

Comment:
Here you could use a more descriptive error message like "Cordova bindings needs Java bindings to be generated first"